app/controllers/admin/projects_controller.rb

```diff
@@ -17,6 +17,7 @@ def destroy
     if Rails.application.config.samson.project_deleted_email
       ProjectMailer.deleted_email(@current_user, project).deliver_later
     end
+    Samson::Hooks.fire(:audit_action, current_user, 'deleted project', project)
```

Comment:
how about instead of adding controller callbacks, which we need to repeat/copy-paste in the api ... we use for example environment_decorator with a new `after_commit :audit` and get the user from Papertrail.whodunnit ?
... would mean 1 module to include in 3-4 models ... possibly could remove hooks too since it could call whatever it needed from inside the plugin like `AuditFoo.record :delete, environment`

roughly:
```Ruby
# app/decorators/environment_decorator.rb

module AuditlogConcern
  def self.included(base)
    base.after_commit -> { |e| AuditLog.record :delete, Papertrail.whodunnit, e, }, on: :destroy
    base.after_commit -> { |e| AuditLog.record :create, Papertrail.whodunnit, e, }, on: :create
    base.after_commit -> { |e| AuditLog.record :update, Papertrail.whodunnit, e, }, on: :update
  end
end

Environment.include AuditlogConcern
```